src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -6076,96 +6097,162 @@ void ObjectMgr::LoadAccessRequirements()
-            }
-        }
+                Field* progression_requirement_row = progression_requirements_results->Fetch();
 
-        if (ar->item2)
-        {
-            ItemTemplate const* pProto = GetItemTemplate(ar->item2);
-            if (!pProto)
-            {
-                sLog->outError("Second item %u does not exist for map %u difficulty %u, removing key requirement.", ar->item2, mapid, difficulty);
-                ar->item2 = 0;
-            }
-        }
+                uint8 requirement_type       = progression_requirement_row[0].GetUInt8();
+                uint32 requirement_id        = progression_requirement_row[1].GetUInt32();
+                std::string requirement_hint = progression_requirement_row[2].GetString();
+                uint8 requirement_faction    = progression_requirement_row[3].GetUInt8();
+                uint32 priorityOrder         = progression_requirement_row[3].IsNull() ? UINT32_MAX : progression_requirement_row[3].GetUInt32();
 
-        if (ar->quest_A)
-        {
-            if (!GetQuestTemplate(ar->quest_A))
-            {
-                sLog->outErrorDb("Required Alliance Quest %u not exist for map %u difficulty %u, remove quest done requirement.", ar->quest_A, mapid, difficulty);
-                ar->quest_A = 0;
-            }
-        }
+                ProgressionRequirement* progression_requirement = new ProgressionRequirement();
+                progression_requirement->id           = requirement_id;
+                progression_requirement->hint         = requirement_hint;
+                progression_requirement->faction      = (TeamId)requirement_faction;
+                progression_requirement->priority     = priorityOrder;
 
-        if (ar->quest_H)
-        {
-            if (!GetQuestTemplate(ar->quest_H))
-            {
-                sLog->outErrorDb("Required Horde Quest %u not exist for map %u difficulty %u, remove quest done requirement.", ar->quest_H, mapid, difficulty);
-                ar->quest_H = 0;
-            }
-        }
+                std::vector<ProgressionRequirement*>* currentRequirementVector = nullptr;
 
-        if (ar->achievement)
-        {
-            if (!sAchievementStore.LookupEntry(ar->achievement))
-            {
-                sLog->outErrorDb("Required Achievement %u not exist for map %u difficulty %u, remove quest done requirement.", ar->achievement, mapid, difficulty);
-                ar->achievement = 0;
-            }
+                switch (requirement_type)
+                {
+                case 0:
+                {
+                    //Achievement
+                    if (!sAchievementStore.LookupEntry(progression_requirement->id))
+                    {
+                        sLog->outErrorDb("Required achievement %u for faction %u does not exist for map %u difficulty %u, remove or fix this achievement requirement.", progression_requirement->id, requirement_faction, mapid, difficulty);
+                        break;
```

Comment:
is it normal break here and another break below? (probably, just asking to make sure)

Reply:
Yes, I am exitring before hand as I do not want the following code to execute